sunpy/map/mapbase.py

```diff
@@ -855,15 +855,17 @@ def pixel_to_world(self, x, y, origin=0):
             A coordinate object representing the output coordinate.
 
         """
-        x, y = self.wcs.wcs_pix2world(x, y, origin)
 
-        # If the wcs is celestial it is output in degress
-        if self.wcs.is_celestial:
-            x = u.Quantity(x, u.deg)
-            y = u.Quantity(y, u.deg)
-        else:
-            x = u.Quantity(x, self.spatial_units[0])
-            y = u.Quantity(y, self.spatial_units[1])
+        # Hold the WCS instance here so we can inspect the output units after
+        # the pix2world call
```

Comment:
This inspection is done outside of the function call?

Reply:
No it's done on the line after the call.